mqtt/mqtt-broker/src/persist.rs

```diff
@@ -571,7 +590,7 @@ pub(crate) mod tests {
     async fn filepersistor_smoketest() {
         let tmp_dir = TempDir::new().unwrap();
         let path = tmp_dir.path().to_owned();
-        let mut persistor = FilePersistor::new(path, ConsolidatedStateFormat::default());
+        let mut persistor = FilePersistor::new(path, VersionedFileFormat::default());
 
         persistor.store(BrokerState::default()).await.unwrap();
```

Comment:
nit: btw, it is better to use some test data in BorkerState, instead of `default()`. When I played with versioned state, I had a peculiar issue where I was writing into a wrong file and reading from the old one. They both returned `default()` state, so the test would pass, but the logic was broken:) using some actual test data revealed the problem.

Reply:
Yea, I think once we have a second version there needs to be a test added for reading an old version, But that can;t be made until there's 2 versions